xscen/aggregate.py

```diff
@@ -0,0 +1,443 @@
+        else:
+            raise ValueError("Delta 'kind' not understood.")
+
+        with xr.set_options(keep_attrs=True):
+            if _kind == "absolute":
+                deltas[f"{vv}_delta_{reference_horizon.replace('-', '_')}"] = (
+                    other_hz[vv] - ref[vv]
+                )
+            else:
+                deltas[f"{vv}_delta_{reference_horizon.replace('-', '_')}"] = (
+                    other_hz[vv] / ref[vv]
+                )
+                deltas[f"{vv}_delta_{reference_horizon.replace('-', '_')}"].attrs[
+                    "units"
+                ] = ""
+
+        # modify attrs and history
+        deltas[f"{vv}_delta_{reference_horizon.replace('-', '_')}"].attrs[
+            "delta_kind"
+        ] = _kind
+        deltas[f"{vv}_delta_{reference_horizon.replace('-', '_')}"].attrs[
+            "delta_reference"
+        ] = reference_horizon
+
+        for a in ["description", "long_name"]:
+            if hasattr(other_hz[vv], a):
+                deltas[f"{vv}_delta_{reference_horizon.replace('-', '_')}"].attrs[
+                    a
+                ] = f"{other_hz[vv].attrs[a]}: {_kind} delta compared to {reference_horizon.replace('-', '_')}."
+
+        new_history = f"[{datetime.datetime.now().strftime('%Y-%m-%d %H:%M:%S')}] {_kind} delta vs. {reference_horizon} - xarray v{xr.__version__}"
+        history = (
+            new_history
+            + " \n "
+            + deltas[f"{vv}_delta_{reference_horizon.replace('-', '_')}"].attrs[
+                "history"
+            ]
+            if "history"
+            in deltas[f"{vv}_delta_{reference_horizon.replace('-', '_')}"].attrs
+            else new_history
+        )
+        deltas[f"{vv}_delta_{reference_horizon.replace('-', '_')}"].attrs[
+            "history"
+        ] = history
+
+    # get back to 1D time
+    deltas = deltas.stack(time=("year", "month", "day"))
+    # rebuild time coord
+    time_coord = [
+        pd.to_datetime(f"{y}, {m}, {d}")
+        for y, m, d in zip(deltas.year.values, deltas.month.values, deltas.day.values)
+    ]
+    deltas = deltas.assign_coords(time=time_coord).transpose("time", "lat", "lon")
+
+    if to_level is not None:
+        deltas.attrs["cat/processing_level"] = to_level
+
+    return deltas
+
+
+def spatial_mean(
+    ds: xr.Dataset,
+    method: str,
+    *,
+    call_clisops: bool = False,
+    region: dict = None,
+    kwargs: dict = None,
+    simplify_tolerance: float = None,
+    to_domain: str = None,
+    to_level: str = None,
+) -> xr.Dataset:
+    """
+    Computes the spatial mean, using a variety of available methods.
+
+    Parameters
+    ----------
+    ds: xr.Dataset
+      Dataset to use for the computation.
+    method: str
+      'mean' will perform a .mean() over the spatial dimensions of the Dataset.
+      'interp_coord' will (optionally) find the region's centroid, then perform a .interp() over the spatial dimensions of the Dataset.
```

Comment:
Not sure why we need this method here ? What do we use it for ?
Also, why the "(optionnally)"? Does it do that when coords are not given ?

Reply:
Since it is much faster than SpatialAverager and potentially more accurate than `mean` depending on the specifics, I figured that there might be cases where we want to use `interp(method="linear")` with the centroid of a region or shapefile.

Also, the "optionally" is because it's in theory possible to feed the coordinates directly to `interp` via the `kwargs` argument.

Reply:
```suggestion
      'interp_coord' will find the region's centroid (if coordinates are not fed through kwargs), then perform a .interp() over the spatial dimensions of the Dataset.
```